widgetssdk/src/main/java/com/glia/widgets/push/notifications/PushNotificationsSetUpUseCase.kt

```diff
@@ -0,0 +1,13 @@
+package com.glia.widgets.push.notifications
+
+internal interface PushNotificationsSetUpUseCase {
+    operator fun invoke(): Boolean
+}
+
+internal class PushNotificationsSetUpUseCaseImpl(private val firebaseAvailabilityCheckManager: FirebaseAvailabilityCheckManager) :
+    PushNotificationsSetUpUseCase {
+    override fun invoke(): Boolean = firebaseAvailabilityCheckManager.run {
+        isFcmDependencyAvailable && isFcmServiceDeclaredInManifest
+    }
+
+}
```

Comment:
I suppose, you will add code that will set up push notifications later, right?

Reply:
No, will request notification permission before the first engagement request If push notifications are set up.

Reply:
In this case, I think that `PushNotificationsSetUpUseCase` and `PushNotificationsSetUpUseCaseImpl` name does not correspond to the actual work it does. I would rename it to something like `PushNotificationseAvailabilityUseCase`.

Also, do we really need 3 classes for this: FirebaseAvailabilityCheckManager, `PushNotificationsSetUpUseCase` and `PushNotificationsSetUpUseCaseImpl`? Maybe it's enough to have one use case class?

Reply:
Makes sense, will rename it and remove the manager layer.